blocks/procedures.js

```diff
@@ -150,6 +152,54 @@ Blockly.Blocks['procedures_defnoreturn'] = {
     // Show or hide the statement input.
     this.setStatements_(xmlElement.getAttribute('statements') !== 'false');
   },
+  /**
+   * Returns the state of this block as a JSON serializable object.
+   * @return {{params: (!Array<{name: string, id: string}>|undefined),
```

Comment:
This needs a `?` before the record type definition, since this function may return null.